packages/aws-cdk/lib/version.ts

```diff
@@ -0,0 +1,98 @@
+import { exec as _exec } from 'child_process';
+import colors = require('colors/safe');
+import { open as _open, stat as _stat } from 'fs';
+import semver = require('semver');
+import { promisify } from 'util';
+import { debug, print, warning } from '../lib/logging';
+
+const ONE_DAY_IN_SECONDS = 1 * 24 * 60 * 60;
+
+const exec = promisify(_exec);
+const open = promisify(_open);
+const stat = promisify(_stat);
+
+export const DISPLAY_VERSION = `${versionNumber()} (build ${commit()})`;
+
+function versionNumber(): string {
+  return require('../package.json').version.replace(/\+[0-9a-f]+$/, '');
+}
+
+function commit(): string {
+  return require('../build-info.json').commit;
+}
+
+export class CacheFile {
+  private readonly file: string;
+
+  // File modify times are accurate only till the second, hence using seconds as precision
+  private readonly ttlSecs: number;
+
+  constructor(file: string, ttlSecs: number) {
+    this.file = file;
+    this.ttlSecs = ttlSecs;
+  }
+
+  public async hasExpired(): Promise<boolean> {
+    try {
+      const lastCheckTime = (await stat(this.file)).mtimeMs;
+      const today = new Date().getTime();
+
+      if ((today - lastCheckTime) / 1000 > this.ttlSecs) { // convert ms to secs
+        return true;
+      }
+      return false;
+    } catch (err) {
+      if (err.code === 'ENOENT') {
+        return true;
+      } else {
+        throw err;
+      }
+    }
+  }
+
+  public async update(): Promise<void> {
+    await open(this.file, 'w');
+  }
+}
+
+// Export for unit testing only.
+// Don't use directly, use displayVersionMessage() instead.
+export async function latestVersionIfHigher(currentVersion: string, cacheFile: CacheFile): Promise<string | null> {
+  if (!(await cacheFile.hasExpired())) {
+    return null;
+  }
+
+  const { stdout, stderr } = await exec(`npm view cdk version`);
+  if (stderr && stderr.trim().length > 0) {
+    debug(`The 'npm view' command generated an error stream with content [${stderr.trim()}]`);
+  }
+  const latestVersion = stdout.trim();
+  if (!semver.valid(latestVersion)) {
+    throw new Error(`npm returned an invalid semver ${latestVersion}`);
+  }
+  const isNewer = semver.gt(latestVersion, currentVersion);
+  await cacheFile.update();
+
+  if (isNewer) {
+    return latestVersion;
+  } else {
+    return null;
+  }
+}
+
+const versionCheckCache = new CacheFile(`${__dirname}/../.LAST_VERSION_CHECK`, ONE_DAY_IN_SECONDS);
+
+export async function displayVersionMessage(): Promise<void> {
+  try {
+    const laterVersion = await latestVersionIfHigher(versionNumber(), versionCheckCache);
+    if (laterVersion) {
+      const fmt = colors.green(laterVersion as string);
+      print('********************************************************');
+      print(`***  Newer version of the CDK is available [${fmt}]  ***`);
+      print(`***  Upgrade now by running "npm up -g cdk"          ***`);
```

Comment:
You haven't done any work to know whether the current CDK version has been installed globally or in the current package's dependency set.

If you're going to suggest commands, please make sure they're correct in all cases.

Reply:
I made this assumption based on the [Getting Started section](https://github.com/awslabs/aws-cdk#getting-started) where there is only one mode of installation - if the user has set it up differently, they would know the alternate way to upgrade.

But you're probably right here, the assumption is probably premature.

Reply:
Dropped recommendation here for how to upgrade.